app/src/main/java/com/pajato/android/gamechat/chat/BaseChatFragment.java

```diff
@@ -254,6 +243,14 @@ protected void processPayload(final View view) {
         }
     }
 
+    /** Do a redisplay to catch potential changes that should be shown in the current view. */
+    protected void redisplay() {
+        FabManager.chat.init(this);
```

Comment:
It strikes me that initializing the FAB manager every time a redisplay change occurs is not the correct approach.  My vision here is that the app is in the onResume lifecycle state with the User just sitting there staring at the screen.  A new message comes in via another User.  That message will likely impact the current screen without the app leaving the onResume state. The FAB manager has already been correctly initialized.  There is no need for this to happen again without leaving the onResume state.